kore/src/Kore/Builtin/KEqual.hs

```diff
@@ -206,3 +203,48 @@ neqKey = "KEQUAL.neq"
 
 iteKey :: IsString s => s
 iteKey = "KEQUAL.ite"
+
+{- | The @KEQUAL.eq@ hooked symbol applied to @term@-type arguments.
+ -}
+data KEqual term =
+    KEqual
+        { symbol :: !Symbol
+        , operand1, operand2 :: !term
+        }
+
+{- | Match the @KEQUAL.eq@ hooked symbol.
+ -}
+matchKEqual :: TermLike variable -> Maybe (KEqual (TermLike variable))
+matchKEqual (App_ symbol [operand1, operand2]) = do
+    hook2 <- (getHook . symbolHook) symbol
+    Monad.guard (hook2 == eqKey)
+    return KEqual { symbol, operand1, operand2 }
+matchKEqual _ = Nothing
+
+termKEquals
+    :: forall variable unifier
+    .  InternalVariable variable
+    => MonadUnify unifier
+    => TermSimplifier variable unifier
+    -> NotSimplifier unifier
+    -> TermLike variable
+    -> TermLike variable
+    -> MaybeT unifier (Pattern variable)
+termKEquals unifyChildren (NotSimplifier notSimplifier) a b =
+    worker a b <|> worker b a
+  where
+    worker termLike1 termLike2
+      | Just KEqual { operand1, operand2 } <- matchKEqual termLike1
+      , isFunctionPattern termLike1
+      , Just value2 <- Bool.matchBool termLike2
+      = lift $ do
+        solution <-
+            fmap OrPattern.fromPatterns
+            <$> Unify.gather
+            $ unifyChildren operand1 operand2
+        finalSolution <-
+            if value2
+                then return solution
+                else notSimplifier SideCondition.top solution
```

Comment:
This is not right when we are doing `\and` simplification: the term should always be `value2` (`true` or `false`) because the unification problem is something like:

```
\and(true, X ==K Y)
```

and `solution` is the result of simplifying `\and(X, Y)`. This is fine for `\equals` simplification, because the unification problem is like:

```
\equals(true, X ==K Y)
```

and `solution` is the result of simplifying `\equals(X, Y)` so that the `term` is always `\top` or `\bottom`. I think it will solve the problem you are seeing with `with-config` tests if we restrict this unification case to `\equals` simplification.